plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftCastPushdown.java

```diff
@@ -0,0 +1,467 @@
+                new CastTestCase("c_varchar_numeric", "smallint", "c_smallint"),
+                new CastTestCase("c_text_numeric", "smallint", "c_smallint"),
+                new CastTestCase("c_nvarchar_numeric", "smallint", "c_smallint"),
+                new CastTestCase("c_varchar_numeric_sign", "smallint", "c_smallint"),
+
+                new CastTestCase("c_real", "integer", "c_integer"),
+                new CastTestCase("c_float4", "integer", "c_integer"),
+                new CastTestCase("c_double_precision", "integer", "c_integer"),
+                new CastTestCase("c_float", "integer", "c_integer"),
+                new CastTestCase("c_float8", "integer", "c_integer"),
+                new CastTestCase("c_varchar_numeric", "integer", "c_integer"),
+                new CastTestCase("c_text_numeric", "integer", "c_integer"),
+                new CastTestCase("c_nvarchar_numeric", "integer", "c_integer"),
+                new CastTestCase("c_varchar_numeric_sign", "integer", "c_integer"),
+
+                new CastTestCase("c_real", "bigint", "c_bigint"),
+                new CastTestCase("c_float4", "bigint", "c_bigint"),
+                new CastTestCase("c_double_precision", "bigint", "c_bigint"),
+                new CastTestCase("c_float", "bigint", "c_bigint"),
+                new CastTestCase("c_float8", "bigint", "c_bigint"),
+                new CastTestCase("c_varchar_numeric", "bigint", "c_bigint"),
+                new CastTestCase("c_text_numeric", "bigint", "c_bigint"),
+                new CastTestCase("c_nvarchar_numeric", "bigint", "c_bigint"),
+                new CastTestCase("c_varchar_numeric_sign", "bigint", "c_bigint"),
+
+                new CastTestCase("c_smallint", "boolean", "c_boolean"),
+                new CastTestCase("c_real", "double", "c_double_precision"),
+                new CastTestCase("c_double_precision", "real", "c_real"),
+                new CastTestCase("c_double_precision", "decimal(10,2)", "c_decimal_10_2"),
+                new CastTestCase("c_char_10", "char(50)", "c_char_50"),
+                new CastTestCase("c_char_10", "char(256)", "c_bpchar"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_varchar_50"),
+                new CastTestCase("c_nvarchar_10", "varchar(50)", "c_nvarchar_50"),
+                new CastTestCase("c_varchar_10", "varchar(50)", "c_text"),
+                new CastTestCase("c_timestamp", "date", "c_date"),
+                new CastTestCase("c_timestamp", "time", "c_time"),
+                new CastTestCase("c_date", "timestamp", "c_timestamp"),
+                new CastTestCase("c_varchar_timestamp", "timestamp", "c_timestamp"),
+                new CastTestCase("c_varchar_timestamptz", "timestamp", "c_timestamp"));
+    }
+
+    @Override
+    protected List<InvalidCastTestCase> invalidCast()
+    {
+        return ImmutableList.of(
+                new InvalidCastTestCase("c_varchar_decimal", "integer"),
```

Comment:
Can we capture the error message on Trino and redshift side ?

Reply:
These data types in `invalidCast` does not get pushdown, So can't capture the error message here. Trino Error message `(.*)Cannot cast (.*) to (.*)` is already mentioned inside `InvalidCastTestCase`. Should be explicitly mention error message here?